### PR TITLE
planner: update plan cache key and value

### PR DESCRIPTION
**Pull request by fzzf678**
### What problem does this PR solve?

Issue Number: ref #36598 

Problem Summary:
At present, binding have not been added to the Cache Key, so a current Key may correspond to multiple plans. After getting a plan, we need to actively judge the reasonableness

### What is changed and how it works?
Update the key and value of plan cache.This is partial job of refactor plan cache.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cache.go [hunk 1]
```diff
@@ -149,13 +150,13 @@ func getGeneralPlan(ctx context.Context, sctx sessionctx.Context, cacheKey kvcac
...
-				sctx.PreparedPlanCache().Delete(cacheKey)
-				break
-			}
+			//if cachedVal.BindSQL != bindSQL {
+			//	// When BindSQL does not match, it means that we have added a new binding,
+			//	// and the original cached plan will be invalid,
+			//	// so the original cached plan can be cleared directly
+			//	sctx.PreparedPlanCache().Delete(cacheKey)
+			//	break
+			//}
```
Comment (qw4990) [inline planner/core/plan_cache.go:159]: We can just remove thes lines.

**Review by qw4990 — APPROVED** [review]

**Review by Reminiscent — APPROVED** [review]

**Pull request merged**